http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/files/FileClientResource.java

```diff
@@ -2,69 +2,114 @@
 
 import java.io.IOException;
 import java.nio.file.Files;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
 
 import javax.inject.Inject;
+import javax.ws.rs.DELETE;
 import javax.ws.rs.GET;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
 
+import org.eclipse.microprofile.config.inject.ConfigProperty;
 import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.resteasy.reactive.RestResponse;
 
+import io.quarkus.logging.Log;
 import io.smallrye.common.annotation.Blocking;
 import io.smallrye.mutiny.Uni;
 
 @Path("/file-client")
 public class FileClientResource {
-    private static final String BIGGER_THAN_TWO_GIGABYTES = OsUtils.SIZE_2049MiB;
-    private final java.nio.file.Path FILE = Files.createTempFile("upload", ".txt").toAbsolutePath();
+    private static final long BIGGER_THAN_TWO_GIGABYTES = OsUtils.SIZE_2049MiB;
+
+    private final java.nio.file.Path file;
+    private final List<java.nio.file.Path> deathRow = new LinkedList<>();
     private final FileClient client;
     private final OsUtils utils;
 
     @Inject
-    public FileClientResource(@RestClient FileClient client) throws IOException {
+    public FileClientResource(@RestClient FileClient client,
+            @ConfigProperty(name = "client.filepath") Optional<String> folder) {
         utils = OsUtils.get();
-        utils.createFile(FILE.toString(), BIGGER_THAN_TWO_GIGABYTES);
+        file = folder
+                .stream()
+                .map(existing -> java.nio.file.Path.of(existing).resolve("upload.txt").toAbsolutePath())
+                .peek(path -> utils.createFile(path, BIGGER_THAN_TWO_GIGABYTES))
+                .findFirst().orElse(null);
         this.client = client;
     }
 
     @GET
     @Path("/client-hash")
     @Blocking
     public Uni<String> calculateHash() {
-        return utils.getSum(FILE.toString());
+        return Uni.createFrom().item(() -> utils.getSum(file));
     }
 
     @GET
     @Path("/hash")
-    public Uni<String> hash() {
+    public String hash() {
```

Comment:
should we add a @Blocking annotation, WDYT?

Reply:
It is already blocking due to return type